src/main/scala/firrtl/transforms/SimpleArithmetic.scala

```diff
@@ -0,0 +1,201 @@
+      case DoPrim(Subw, Seq(e, lit@SIntLiteral(n, IntWidth(w))), consts: Seq[BigInt], tpe) if (n < 0) && notMax(n, w) =>
+        DoPrim(Addw, Seq(e, lit.copy(n * -1)), consts, tpe)
+      case _ => e
+    }
+  }
+
+  /** Mapping from references to the [[firrtl.ir.Expression Expression]]s that drive them */
+  type Netlist = mutable.HashMap[WrappedExpression, Expression]
+
+  /** Recursively replace [[WRef]]s with new [[Expression]]s
+    *
+    * @param netlist a '''mutable''' HashMap mapping references to [[firrtl.ir.DefNode DefNode]]s to their connected
+    * [[firrtl.ir.Expression Expression]]s. It is '''not''' mutated in this function
+    * @param expr the Expression being transformed
+    * @return Returns expr with Boolean Expression inlined
+    */
+  def onExpr(netlist: Netlist, symbolTable: Map[WrappedExpression, Int])(expr: Expression): Expression = {
+    expr.map(onExpr(netlist, symbolTable)) match {
+      case DoPrim(op, args, _, _) if isSimpleArithmeticExpr(op) => replaceSimple(expr)
+      case other => other
+    }
+  }
+
+  /** Inline bits in a Statement
+    *
+    * @param netlist a '''mutable''' HashMap mapping references to [[firrtl.ir.DefNode DefNode]]s to their connected
+     * [[firrtl.ir.Expression Expression]]s. This function '''will''' mutate it if stmt is a [[firrtl.ir.DefNode
+    * DefNode]] with a Temporary name and a value that is a [[PrimOp]] Bits
+    * @param stmt the Statement being searched for nodes and transformed
+    * @return Returns stmt with Bits inlined
+    */
+  def onStmt(netlist: Netlist, symbolTable: Map[WrappedExpression, Int])(stmt: Statement): Statement = {
+    println(s"stmt = ${stmt.serialize}")
+    stmt.map(onStmt(netlist, symbolTable)).map(onExpr(netlist, symbolTable))
+  }
+
+  /** Replaces bits in a Module */
+  def onMod(mod: DefModule,
+    netlist: Netlist,
+    symbolTable: Map[WrappedExpression, Int]): DefModule =
+    mod.map(onStmt(netlist, symbolTable))
+}
+
+/**
+  * {{{
+  *   circuit Top :
+  *   module Top :
+  *     input a : SInt<2>
+  *     output y : SInt<2>
+  *     y <= add(a, SInt(-1))
+  * }}}
+  *
+  * converts to
+  *
+  * {{{
+  *   circuit Top :
+  *   module Top :
+  *     input a : SInt<2>
+  *     output y : SInt<2>
+  *     y <= sub(a, SInt(1))
+  * }}}
+  *
+  * ________________________
+  *
+  * * {{{
+  *   *   circuit Top :
+  *   *   module Top :
+  *   *     input a : SInt<2>
+  *   *     output y : SInt<2>
+  *   *     y <= sub(a, SInt(-1))
+  *   * }}}
+  * *
+  * * converts to
+  * *
+  * * {{{
+  *   *   circuit Top :
+  *   *   module Top :
+  *   *     input a : SInt<2>
+  *   *     output y : SInt<2>
+  *   *     y <= add(a, SInt(1))
+  *   * }}}
+   */
+class SimpleArithmetic extends Transform {
+  def inputForm = LowForm
+  def outputForm = LowForm
+
+  import SimpleArithmetic._
+
+  def makeModuleInfo(c: Circuit): (Map[String, Map[String, String]], Seq[String], InstanceGraph, Map[String, DefModule]) = {
```

Comment:
I don't think you call this function - copy pasta?

Reply:
Yes